src/app/issuer/components/ai-assistant/ai-assistant.component.ts

```diff
@@ -0,0 +1,161 @@
+import { AbstractControl, FormsModule, Validators } from '@angular/forms';
+import { TranslateModule, TranslateService } from '@ngx-translate/core';
+import { ApiSkill } from '~/common/model/ai-skills.model';
+import { AiSkillsService } from '~/common/services/ai-skills.service';
+import { MessageService } from '~/common/services/message.service';
+import { typedFormGroup } from '~/common/util/typed-forms';
+import { CompetencyAccordionComponent } from '~/components/accordion.component';
+import { AltchaComponent } from '~/components/altcha.component';
+import { OebInputComponent } from '~/components/input.component';
+import { OebButtonComponent } from '~/components/oeb-button.component';
+import { OebCheckboxComponent } from '~/components/oeb-checkbox.component';
+
+@Component({
+	selector: 'ai-assistant',
+	templateUrl: './ai-assistant.component.html',
+	standalone: true,
+	imports: [
+		TranslateModule,
+		OebButtonComponent,
+		FormsModule,
+		CommonModule,
+		AltchaComponent,
+		OebInputComponent,
+		OebCheckboxComponent,
+		CompetencyAccordionComponent,
+	],
+})
+export class AiAssistantComponent implements AfterViewInit {
+	aiCompetenciesLoading = false;
+	aiCompetenciesDescription: string = '';
+	aiCompetenciesSuggestions: ApiSkill[] = [];
+	suggestCompetenciesText = this.translate.instant('CreateBadge.suggestCompetencies');
+	detailedDescription = this.translate.instant('CreateBadge.detailedDescription');
+	showBadgeDemo = false;
+
+	@ViewChild('altcha') altcha: AltchaComponent;
+	altchaValue = '';
+
+	@ViewChild('top') top: ElementRef<HTMLElement>;
+
+	positiveInteger(control: AbstractControl) {
+		const val = parseInt(control.value, 10);
+		if (isNaN(val) || val < 1) {
+			return { expires_amount: 'Must be a positive integer' };
+		}
+	}
+
+	positiveIntegerOrNull(control: AbstractControl) {
+		const val = parseFloat(control.value);
+
+		if (isNaN(val)) {
+			return { emptyField: 'Das Feld darf nicht leer sein.' };
+		}
+		if (!Number.isInteger(val) || val < 0) {
+			return { negativeDuration: 'Bitte geben Sie eine positive Zahl oder 0 ein.' };
+		}
+	}
+
+	aiForm = typedFormGroup().addArray(
+		'aiCompetencies',
+		typedFormGroup()
+			.addControl('selected', false)
+			.addControl('hours', '1', [this.positiveIntegerOrNull, Validators.max(999)])
+			.addControl('minutes', '0', [this.positiveIntegerOrNull, Validators.max(59)]),
+	);
+
+	requiredError = this.translate.instant('CreateBadge.requiredError');
+
+	constructor(
+		protected aiSkillsService: AiSkillsService,
+		private translate: TranslateService,
+		private messageService: MessageService,
+		private changeDetectorRef: ChangeDetectorRef,
+	) {}
+
+	ngAfterViewInit(): void {
+		this.altcha.valueEvent.subscribe((value) => {
+			if (value) {
+				this.altchaValue = value;
+				// FIXME: not sure why this is needed, but angular does not update the view otherwise
+				this.changeDetectorRef.detectChanges();
```

Comment:
I suppose thats because updating the ViewChild actually doesn't trigger a change detection cycle and only marks the property to be checked.
No real way around that, I suppose :shrug: 

Might as well remove the //FIXME right?